src/front/glsl/functions.rs

```diff
@@ -1058,6 +1058,71 @@ impl Parser {
         });
     }
 
+    fn arg_type_walker(
+        &self,
+        name: Option<String>,
+        binding: crate::Binding,
+        pointer: Handle<Expression>,
+        ty: Handle<Type>,
+        expressions: &mut Arena<Expression>,
+        f: &mut impl FnMut(
```

Comment:
could this be `mut f: impl FnMut`?

Reply:
That was my original idea but as the function is recursive it needs to pass the function to itself and passing it as a reference when recursing causes infinite type recursion.